game/services.py

```
@@ -20,34 +20,113 @@
-    ActiveGame.objects.update_or_create(
-        session_key=request.session.session_key,
-        defaults={
-            "user": request.user if request.user.is_authenticated else None,
-            "status": "active",
-            "last_active": timezone.now(),
-        },
-    )
+    active_game, _, version = load_game_state_helper(request)
+    success_save, _ = save_game_state_helper(request, active_game, game_state, version)
+    if not success_save:
+        raise ValueError("Conflict: Stale game state.")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)